crates/ark/src/interface.rs

```diff
@@ -1733,9 +1733,14 @@ impl RMain {
         RHelp::is_help_url(url, port)
     }
 
-    fn send_lsp_notification(&self, event: KernelNotification) {
+    fn send_lsp_notification(&mut self, event: KernelNotification) {
         if let Some(ref tx) = self.lsp_events_tx {
-            tx.send(Event::Kernel(event)).unwrap();
+            if let Err(err) = tx.send(Event::Kernel(event)) {
+                log::error!("Failed to send LSP notification: {err:?}");
+                self.lsp_events_tx = None;
+            }
+        } else {
+            log::warn!("Not sending LSP notification (no connection): {event:?}");
```

Comment:
This would warn in Jupyter apps. We should probably keep treating the absence of connected LSPs as a normal state.

Reply:
that's fair!